src/blockdata/transaction.rs

```diff
@@ -220,12 +220,80 @@ impl Default for TxIn {
         TxIn {
             previous_output: OutPoint::default(),
             script_sig: Script::new(),
-            sequence: u32::max_value(),
+            sequence: Sequence::MAX,
             witness: Witness::default(),
         }
     }
 }
 
+/// Bitcoin transaction input sequencing number.
+/// 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Sequence(pub u32);
+
+impl Sequence {
+    /// The maximum allowable ssequence number
+    pub const MAX: Self = Sequence(MAX_SEQUENCE);
+    
+    const RBF_SIGNAL: u32 = 0xFFFFFFFE;
+    const DISABLE_FLAG_MASK: u32 = 0x80000000;
+    const LOCK_TYPE_MASK: u32 = 0x00400000;
+    
+    /// Retuns `true` if the sequence number is 0xffffffff.
+    pub fn is_final(&self) -> bool {
+        *self == Sequence::MAX
+    }
+
+    /// Returns `true` if the sequence number is less than 0xfffffffe which
+    /// indicates that the transaction is opted-in to BIP125 replace-by-fee.
+    pub fn is_rbf(&self) -> bool {
+        self.0 < Sequence::RBF_SIGNAL
+    }
+
+    /// Returns `true` if the sequence has a relative timelock.
+    pub fn is_timelock(&self) -> bool {
+        self.0 & Sequence::DISABLE_FLAG_MASK == 0
+    }
+
+    /// Returns true if the sequence number encodes a block-based lock.
+    pub fn is_height_locked(&self) -> bool {
+        self.is_timelock() & (self.0 & Sequence::LOCK_TYPE_MASK == 0)
+    }
+
+    /// Returns true if the sequene number encodes a time based lock.
+    pub fn is_time_locked(&self) -> bool {
+        self.is_timelock() & (self.0 & Sequence::LOCK_TYPE_MASK > 0)
+    }
+    
+    /// Create a relative timelock using block height.
+    pub fn from_height(height: u16) -> Self {
+        Sequence::from_consensus(height as u32)
+    }
+
+    /// Create a relative timelock using time units where each unit is equivalent to 512 seconds.
+    pub fn from_timeunits(units: u16) -> Self {
+        Sequence::from_consensus(units as u32 | Sequence::LOCK_TYPE_MASK)
```

Comment:
Please use `from` or `into` instead of `as`.

Reply:
For your learning, the reason is that using `units.into()` is guaranteed lossless, saving devs from having to glance at the types when reading this code to make sure that there is not rounding being done. (Although admittedly this is just a clean-coding habit since we have clippy in CI now it would catch rounding.)